jmxtrans-cluster/src/main/java/com/googlecode/jmxtrans/cluster/zookeeper/ZookeeperJvmHandler.java

```diff
@@ -0,0 +1,300 @@
+package com.googlecode.jmxtrans.cluster.zookeeper;
+
+import com.google.common.base.Optional;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.*;
+import org.apache.curator.framework.recipes.leader.LeaderLatch;
+import org.apache.curator.framework.recipes.leader.LeaderLatchListener;
+import org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * ZookeeperJvmHandler.
+ *
+ * @author Tibor Kulcsar
+ * @since <pre>May 17, 2016</pre>
+ */
+public class ZookeeperJvmHandler  {
```

Comment:
I need to take more time to review this class. It looks like it needs to be threadsafe, but I suspect it is not at the moment.